### PR TITLE
tests for QualityScoreDistribution

### LINKED ISSUE #408: add tests for QualityScoreDistribution
need to enumerate test cases and implement each

### PR DESCRIPTION
**Pull request by akiezun**
thanks to @dekling 
fixed #408

### CODE REVIEW

File: src/test/java/org/broadinstitute/hellbender/tools/IntegrationTestSpec.java [hunk 1]
```diff
@@ -168,9 +170,24 @@ public static void assertMatchingFiles(List<File> resultFiles, List<String> expe
...
+        compareTextFiles(resultFile, expectedFile, null);
+    }
+
+    /**
+     * Compares two text files and ignores all linest that start with the comment prefix.
+     */
+    public static void compareTextFiles(final File resultFile, final File expectedFile, final String commentPrefix) throws IOException {
+        final Predicate<? super String> doesNotStartWithComment;
+        if (commentPrefix == null){
+            doesNotStartWithComment = s -> false;
```
Comment (droazen) [inline src/test/java/org/broadinstitute/hellbender/tools/IntegrationTestSpec.java]: This branch appears to have the same bug as @lbergelson identified in https://github.com/broadinstitute/hellbender/pull/636 -- could you please patch it before I review?

**Comment by droazen** [comment]
Back to @akiezun to patch this branch before I review.

**Comment by akiezun** [comment]
i'll wait for #636  to be merged and rebase on it

**Comment by akiezun** [comment]
back to @droazen then

File: src/test/java/org/broadinstitute/hellbender/tools/picard/analysis/QualityScoreDistributionIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    private static final File TEST_DATA_DIR = new File(getTestDataDir(), "picard/analysis/QualityScoreDistribution");
+
+    @Test
+    public void test() throws IOException {
+        final File input = new File(TEST_DATA_DIR, "first5000a.bam");
+        final File expectedFile = new File(TEST_DATA_DIR, "qualscoredist.txt");
+        final File outfile = BaseTest.createTempFile("testQualityScoreDistribution", ".metrics");
+        final File pdf = BaseTest.createTempFile("testQualityScoreDistribution", ".pdf");
+        outfile.deleteOnExit();
+        pdf.deleteOnExit();
```
Comment (droazen) [inline src/test/java/org/broadinstitute/hellbender/tools/picard/analysis/QualityScoreDistributionIntegrationTest.java]: No need to set `deleteOnExit()` -- `BaseTest.createTempFile()` does this for you.

**Comment by droazen** [comment]
Back to @akiezun -- one comment, then good to merge.

**Pull request merged**